### DISCUSSION TITLE
We have trouble receiving subscription events when using an express server, but have it working fine with a fastify server.

### DISCUSSION THREAD

**Discussion by DevRichter** [enisdenjo/graphql-sse#109] [Q&A] [answered]
We have a federated wundergraph, that routes all subgraph requests. We have one service, which runs on fastify, where the `graphql-sse` package works fine. Events gets published and sent to the correct clients, with the correct data. We have 2 other services which run on express. Both services have trouble actually calling the subscription at all. The events get published in the ioredis, and as the asyncIterator is the exact same as in the fastify service, i expect the events to reach my client - which they dont. We use the exact same code, except for
// Express
![image](https://github.com/user-attachments/assets/57b7f797-a6a8-4670-9684-7c471b5668b8)
![image](https://github.com/user-attachments/assets/16c58e7c-3480-47a3-9207-dda290344b9e)

// Fastify (ignore the import error its just an ide issue)
![image](https://github.com/user-attachments/assets/59f45207-f4fa-4d60-9d63-1e1d030d5794)
![image](https://github.com/user-attachments/assets/dbfc1455-9ff2-40d3-b5f4-f72e595eff41)

are there any prerequisites for express servers, that we are maybe missing? Any ideas what to look for? We are struggling with this for quite some time

Edit: ive changed the use of the package to exactly how it is in the guild documentation -> `import { createHandler } from 'graphql-sse'; ` instead of the path to express. Now i receive this err ->

![image](https://github.com/user-attachments/assets/f13f0cba-3084-40f3-97c1-a80a48b22698)

maybe this helps?

**Comment by enisdenjo** [comment, marked as answer]
Hmm, that's very weird. graphql-sse has thorough tests for each of the frameworks' integrations ([see here](https://github.com/enisdenjo/graphql-sse/blob/master/tests/use.test.ts)), the backbone of all them tests is a functioning handler and they all pass. 🤔 

I'd be willing to take a look if you can create me a repro or a thorough replication steps. Sorry for the inconvenience, I hope we solve this soon!

About your EDIT, you cant use the `createHandler` directly from the root because it's server agnostic and it does not adapt to `express`. You must use the `express` specific handler as [showcased in the documentation for `express`](https://the-guild.dev/graphql/sse/get-started#with-express).

**Comment by DevRichter** [comment]
Okay, ive noticed a few problems, solved them, now it works. The package is working fine. Thanks for the great work.
